lib/Retriever.js

```diff
@@ -32,16 +32,25 @@ class Retriever {
      * Retrieve metadata of specified types into local file system and Retriever.metadata
      *
      * @param {string[]} metadataTypes String list of metadata types to retrieve
-     * @param {string} [name] name of Metadata to retrieve (in case of templating)
+     * @param {string[]} [nameOrKey] name of Metadata to retrieveAsTemplate or list of keys for normal retrieval
      * @param {TYPE.TemplateMap} [templateVariables] Object of values which can be replaced (in case of templating)
      * @param {boolean} [changelogOnly] skip saving, only create json in memory
      * @returns {Promise.<TYPE.MultiMetadataTypeList>} Promise of a list of retrieved items grouped by type {automation:[...], query:[...]}
      */
-    async retrieve(metadataTypes, name, templateVariables, changelogOnly) {
+    async retrieve(metadataTypes, nameOrKey, templateVariables, changelogOnly) {
```

Comment:
do we still need to keep name? I see in other places you reference Key only (not name)

Reply:
name for retrieveAsTemplate for as long as we do not want to deprecate that.


side note: pretty sure you chose name because the REST apis allow filtering by name - but not by key or anything else